### PR TITLE
perf: take Tree::new items by reference

### PR DESCRIPTION
**Pull request by sandercm**
Currently you are forced to copy the items into the tree. With this change you can avoid this overhead. It works for my use case but if I missed something please let me know.

### CODE REVIEW

**Comment by EdJoPaTo** [comment]
The idea of a `TreeItem` is to only keep a reference to its content rather than moving the full content into the item. That's why the `TreeItem` has a reference in there: `TreeItem<'a, Identifier>`. It's a reference to the original content. `Text<'a>` does the same, keeping the reference to the original content.

As the `TreeItem` should be rather small this way it shouldn't improve on being a reference again.

I think benchmarks would be interesting to find out whether this is a bottleneck or not.

**Comment by EdJoPaTo** [comment]
I hope something like this is not necessary, and I prefer improving the performance rather than using something like this. ([More detailed answer in the related issue](https://github.com/EdJoPaTo/tui-rs-tree-widget/issues/31#issuecomment-2088770092)). So I would close this for now. Or do you have other thoughts here?

**Comment by sandercm** [comment]
I agree that improving the performance of TreeItem seems like a good idea. But I'm not sure why you don't like this approach of passing the items by reference. Even if the performance overhead is minimal when you are dealing with large json objects, you basically have deeply recursive datastructres that require a full traversal of the vec to copy it even if the indiviual copies are super fast you still end up with O(n) operations vs O(1) if you pass a reference. For smaller json objects it's basically unnoticable but for larger json objects the performance gain is huge. 

In my case my fps went x15-20 simply making Tree take a reference (for json ~11MB), so I would be interested in hearing why you'd prefer to not use them?

File: src/lib.rs [hunk 1]
```diff
@@ -52,7 +52,7 @@ pub use crate::tree_state::TreeState;
 /// ```
 #[derive(Debug, Clone)]
 pub struct Tree<'a, Identifier> {
-    items: Vec<TreeItem<'a, Identifier>>,
+    items: &'a Vec<TreeItem<'a, Identifier>>,
```
Comment (EdJoPaTo) [inline src/lib.rs:55]: Just noticed this while glancing over this. When there is a reference, a slice would be better than a Vec.

```suggestion
    items: &'a [TreeItem<'a, Identifier>],
```

**Review by EdJoPaTo — COMMENTED** [review]
I am thinking about a release in between the current v0.19.0 and the upcoming `TreeData` trait changes (#34). The `TreeData` allows for a lot more flexibility and is also taken by reference. It improves performance by a different approach and includes an example for JSON, which is way more performant then due to not generating items that aren't rendered.

To reduce the friction a release in between might be helpful and given that the `TreeData` takes a reference it might be a good idea to include this.

**Comment by sandercm** [comment]
Hi, I haven't had time to continue working on my side projects until now but these changes are great!

**Pull request merged**